src/nvim/if_cscope.c

```diff
@@ -1420,12 +1436,12 @@ static char *cs_manage_matches(char **matches, char **contexts, int totmatches,
     break;
   case Free:
     if (mp != NULL) {
-      if (cnt > 0)
-        while (cnt--) {
-          xfree(mp[cnt]);
-          if (cp != NULL)
-            xfree(cp[cnt]);
-        }
+      while (cnt > 0) {
+        cnt--;
```

Comment:
Yes, good point. I don't remember why I moved it out in the first place. I was probably worried about underflow but when it reaches 0 the loop will stop anyway.